sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventDataBatch.java

```diff
@@ -43,12 +60,25 @@
         this.eventBytes = new byte[maxMessageSize];
     }
 
+    /**
+     * Gets the number of {@link EventData events} in the batch.
+     *
+     * @return The number of {@link EventData events} in the batch.
+     */
     int getSize() {
```

Comment:
I worry that the the name is potentially confusing;  size could be interpreted as the size of the batch, in bytes, or the number of events in the batch.   My brain went to the first and had trouble reconciling with the code.  Maybe consider something like `numberOfEvents` or `countOfEvents`?

Reply:
Yeah, me too, when I first saw this class. `getNumberOfEvents` sounds like the winner.

Reply:
Track 1 .Net SDK has this has `count`, so maybe `getCount`? 
Also, what is the use case for exposing count?

I can see the reason for size which @jsquire mentioned in https://github.com/Azure/azure-sdk-for-java/pull/4403/files#r302972436, but of what use is the count? If the purpose of an "event batch" is to ensure that we improve perf by batching multiple events under a size limit, then count shouldn't be the driving factor?

Reply:
We'll need at least an internal count or other way to understand if a batch is empty.   Size is not likely sufficient, as that would require understanding the envelope size (which isn't stable in .NET, it changes based on something that I don't remember).

I would ask the inverse question:  is there a reason NOT to offer a count, since it's something that batch should understand and an expected member of the intrinsic collection types of each language?

Reply:
Definite yes to the need for an internal count.
Like the idea of using count to determine emptiness

Reply:
To be fair, this is not public API. I don't know if that is intentional or not, but given this, naming is less critical.

Reply:
In our team channel, we've discussed.. it's going to be public now.